src/System.Linq.Expressions/src/System/Linq/Expressions/MemberMemberBinding.cs

```diff
@@ -40,10 +40,15 @@ internal MemberMemberBinding(MemberInfo member, ReadOnlyCollection<MemberBinding
         /// <returns>This expression if no children changed, or an expression with the updated children.</returns>
         public MemberMemberBinding Update(IEnumerable<MemberBinding> bindings)
         {
-            if (bindings == Bindings)
+            if (bindings != null)
             {
-                return this;
+                bindings = bindings as ICollection<MemberBinding> ?? bindings.ToReadOnly();
+                if (ExpressionUtils.SameElements(bindings, Bindings))
```

Comment:
This looks like a repeating pattern snapping an enumerable into a readonly. Could it be moved into SameElements? Then the typecheck can also be done after the reference comparison (which we hope will often result in easy out)